relay_rpc/src/rpc/msg_id.rs

```diff
@@ -0,0 +1,27 @@
+use {
+    crate::rpc,
+    sha2::{Digest, Sha256},
+    std::sync::Arc,
+};
+
+pub trait MsgId {
+    fn msg_id(&self) -> Arc<str>;
+}
+
+impl MsgId for rpc::Publish {
+    fn msg_id(&self) -> Arc<str> {
+        let msg_id = Sha256::new()
+            .chain_update(self.message.as_ref().as_bytes())
+            .finalize();
+        format!("{msg_id:x}").into()
```

Comment:
This conversion to `Arc<str>` here is bothering me. Let's return `String` and leave it up to the consumer whether to convert it into a shared `Arc<str>` or not. The problem here is that this conversion causes another allocation, which may be totally unnecessary in some cases. But when it is necessary, the consumer of this API can convert it themselves.

Reply:
Done